custom-login/src/Home.jsx

```diff
@@ -10,88 +10,90 @@
  * See the License for the specific language governing permissions and limitations under the License.
  */
 
-import { withAuth } from '@okta/okta-react';
-import React, { Component } from 'react';
+import { useOktaAuth } from '@okta/okta-react';
+import React, { useState, useEffect } from 'react';
 import { Button, Header } from 'semantic-ui-react';
-import { checkAuthentication } from './helpers';
 
-export default withAuth(class Home extends Component {
-  constructor(props) {
-    super(props);
-    this.state = { authenticated: null, userinfo: null };
-    this.checkAuthentication = checkAuthentication.bind(this);
-    this.login = this.login.bind(this);
-  }
+const Home = () => {
+  const { authState, authService } = useOktaAuth();
+  const [userInfo, setUserInfo] = useState(null);
 
-  async componentDidMount() {
-    this.checkAuthentication();
-  }
+  useEffect(() => {
+    if (!authState.isAuthenticated) {
+      // When user isn't authenticated, forget any user info
+      setUserInfo(null);
+    } else {
+      authService.getUser().then((info) => {
+        setUserInfo(info);
+      });
+    }
+  }, [authState, authService]); // Update if authState changes
 
-  async componentDidUpdate() {
-    this.checkAuthentication();
-  }
+  const login = async () => {
+    authService.login('/');
+  };
 
-  async login() {
-    this.props.auth.login('/');
-  }
+  const resourceServerExamples = [
+    {
+      label: 'Node/Express Resource Server Example',
+      url: 'https://github.com/okta/samples-nodejs-express-4/tree/master/resource-server',
+    },
+    {
+      label: 'Java/Spring MVC Resource Server Example',
+      url: 'https://github.com/okta/samples-java-spring-mvc/tree/master/resource-server',
+    },
+  ];
 
-  render() {
-    const resourceServerExamples = [
-      {
-        label: 'Node/Express Resource Server Example',
-        url: 'https://github.com/okta/samples-nodejs-express-4/tree/master/resource-server',
-      },
-      {
-        label: 'Java/Spring MVC Resource Server Example',
-        url: 'https://github.com/okta/samples-java-spring-mvc/tree/master/resource-server',
-      },
-    ];
+  if (authState.isPending) {
+    return null; // wait for authState to be figured out
```

Comment:
Maybe we can render "Loading..." or something here. Would hate to get  bug report that says "I just see a blank page". Would rather "It is stuck on Loading..."